packages/shared/src/hooks/useCompanionTrigger.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useCallback, useContext } from 'react';
+import { useFeatureIsOn } from '../components/GrowthBookProvider';
+import { FeedPostClick } from './feed/useFeedOnPostClick';
+import { Post } from '../graphql/posts';
+import { Features } from '../lib/featureManagement';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+
+import { useExtensionPermission } from './useExtensionPermission';
+import { useLazyModal } from './useLazyModal';
+import { LazyModal } from '../components/modals/common/types';
+import { useActions } from './useActions';
+import { ActionType } from '../graphql/actions';
+
+type CompanionTriggerProps = {
+  post: Post;
+  index?: number;
+  row?: number;
+  column?: number;
+};
+
+type OpenArticle = (data?: CompanionTriggerProps) => () => Promise<void>;
+
+export type CompanionTrigger = {
+  onFeedArticleClick: FeedPostClick;
+  onPostArticleClick: PostClick;
+  activateCompanion: () => void;
+  openArticle: OpenArticle;
+  toggleOpen: (value: boolean, data?: CompanionTriggerProps) => void;
+};
+
+type PostClick = (e: React.MouseEvent, data: { post: Post }) => Promise<void>;
+
+export default function useCompanionTrigger(
+  customPostClickHandler: (
+    post: Post,
+    index?: number,
+    row?: number,
+    column?: number,
+  ) => Promise<void>,
+): CompanionTrigger {
+  const isExtension = process.env.TARGET_BROWSER;
+  const { trackEvent } = useContext(AnalyticsContext);
+  const { closeModal, openModal } = useLazyModal();
+  const { isActionsFetched, checkHasCompleted, completeAction } = useActions();
+
+  const featureEnabled = useFeatureIsOn(
+    Features.EngagementLoopJuly2023Companion,
+  );
+
+  const { requestContentScripts, useContentScriptStatus } =
+    useExtensionPermission({
+      origin: 'companion modal permission button',
+    });
+
+  const { contentScriptGranted } = useContentScriptStatus();
+  const alreadyCompleted = checkHasCompleted(
+    ActionType.EngagementLoopJuly2023CompanionModal,
+  );
+
+  const handleCompleteAction = useCallback(() => {
+    if (isActionsFetched)
+      completeAction(ActionType.EngagementLoopJuly2023CompanionModal);
+  }, [isActionsFetched]);
+
+  const articleClickHandler = useCallback(
+    async (
+      e: React.MouseEvent,
+      { post, index, row, column }: Partial<CompanionTriggerProps> = {},
+    ) => {
+      // the check whether the user is logged in is done on the GrowthBook side
+      // the feature flag is enabled only for logged-in users
+      // -- check that this is correct
+      if (
+        !isExtension ||
+        !featureEnabled ||
+        alreadyCompleted ||
+        contentScriptGranted
+      ) {
+        await customPostClickHandler(post, index, row, column);
+      } else {
+        e.preventDefault();
```

Comment:
Suggestion is alternative maybe we can only define the useFeatureIsOn here?
I know it's a bit anti-pattern, but that would make sure everything keeps working.

Reply:
I've updated this to check for the feature flag as the last check -- had to switch to using the growthbook object instead of the hook -- let me know what you think